--- procps-3.3.17.orig/pmap.c	2021-02-09 17:11:25.000000000 +0700
+++ procps-3.3.17/pmap.c	2021-05-11 23:58:45.000000000 +0700
@@ -24,7 +24,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef __ANDROID__
 #include <sys/ipc.h>
 #include <sys/shm.h>
+#endif
 #include <sys/stat.h>
 #include <sys/types.h>
@@ -150,6 +152,7 @@
 
 static void discover_shm_minor(void)
 {
+#ifndef __ANDROID__
 	void *addr;
 	int shmid;
 	char mapbuf_b[256];
@@ -206,6 +209,7 @@
 		perror(_("shared memory remove"));
 
 	return;
+#endif
 }
 
 static const char *mapping_name(const proc_t * p, unsigned KLONG addr,
diff -ru procps-3.3.17.orig/proc/escape.c procps-3.3.17/proc/escape.c
