--- procps-3.3.17.orig/sysctl.c	2021-02-09 17:11:25.000000000 +0700
+++ procps-3.3.17/sysctl.c	2021-05-12 00:03:30.000000000 +0700
@@ -29,7 +29,9 @@
 #include <dirent.h>
 #include <errno.h>
 #include <getopt.h>
+#ifndef __ANDROID__
 #include <glob.h>
+#endif
 #include <libgen.h>
 #include <limits.h>
 #include <regex.h>
@@ -515,6 +517,9 @@
  */
 static int Preload(const char *restrict const filename)
 {
+#ifdef __ANDROID__
+	return -1;
+#else
 	FILE *fp;
 	char *t;
 	int n = 0;
@@ -604,6 +609,7 @@
 	}
 out:
 	return rc;
+#endif
 }
 
 struct pair {
diff -ru procps-3.3.17.orig/top/top.c procps-3.3.17/top/top.c
