--- moreutils-0.65-orig/is_utf8/Makefile	2021-05-11 19:28:55.304632892 +0000
+++ moreutils-0.65/is_utf8/Makefile	2021-05-11 19:29:35.192101142 +0000
@@ -37,8 +37,8 @@
 SONAME = $(LINKERNAME).$(VERSION)
 REALNAME = $(SONAME).$(MINOR).$(RELEASE)
 
-CC = gcc
-CFLAGS ?= -O3 -Wextra -Wall -ansi -Wstrict-prototypes
+CC ?= gcc
+CFLAGS += -O3 -Wextra -Wall -ansi -Wstrict-prototypes
 
 $(NAME): $(OBJ)
 	$(CC) $(CFLAGS) -o $(NAME) $(OBJ) $(LDFLAGS)
