--- ./tool/rbinstall.rb.orig	2021-05-27 15:48:07.687418031 +0000
+++ ./tool/rbinstall.rb	2021-05-27 15:53:11.603655327 +0000
@@ -897,7 +897,7 @@
 end
 
 def install_default_gem(dir, srcdir)
-  gem_dir = Gem.default_dir
+  gem_dir = "@TERMUX_PREFIX@/lib/ruby/gems/" + RbConfig::CONFIG['ruby_version']
   install_dir = with_destdir(gem_dir)
   prepare "default gems from #{dir}", gem_dir
   makedirs(Gem.ensure_default_gem_subdirectories(install_dir, $dir_mode).map {|d| File.join(gem_dir, d)})
@@ -943,7 +943,7 @@
 end
 
 install?(:ext, :comm, :gem, :'bundled-gems') do
-  gem_dir = Gem.default_dir
+  gem_dir = "@TERMUX_PREFIX@/lib/ruby/gems/" + RbConfig::CONFIG['ruby_version'
   install_dir = with_destdir(gem_dir)
   prepare "bundled gems", gem_dir
   makedirs(Gem.ensure_gem_subdirectories(install_dir, $dir_mode).map {|d| File.join(gem_dir, d)})
