--- ./library/std/src/sys/unix/os.rs.orig	2021-05-27 17:34:04.568708230 +0000
+++ ./library/std/src/sys/unix/os.rs	2021-05-27 17:34:25.447690344 +0000
@@ -578,7 +578,7 @@
 pub fn temp_dir() -> PathBuf {
     crate::env::var_os("TMPDIR").map(PathBuf::from).unwrap_or_else(|| {
         if cfg!(target_os = "android") {
-            PathBuf::from("/data/local/tmp")
+            PathBuf::from("@TERMUX_PREFIX@/data/local/tmp")
         } else {
             PathBuf::from("/tmp")
         }
