--- a/operating_system.rb	2020-02-06 11:43:07.000000000 +0000
+++ b/operating_system.rb	2021-05-15 13:21:06.175292976 +0000
@@ -47,8 +47,13 @@
 
   if RUBY_VERSION >= '2.7'
     class Gem::Specification
-      alias :upstream_rubyforge_project= :rubyforge_project=
-      def rubyforge_project=(x)
+      begin
+        alias :upstream_rubyforge_project= :rubyforge_project=
+        def rubyforge_project=(x)
+        end
+      rescue NameError
+        # rubyforge_project= was removed in RubyGems 3.1.4
+        # If method rubyforge_project= is not present, that's ok
       end
     end
   end
